spec/lib/cyclonedx/report_ruby_gems_cyclonedx_spec.rb

```diff
@@ -0,0 +1,155 @@
+require_relative '../../spec_helper'
+require 'json'
+
+describe Cyclonedx::ReportRubyGems do
+  describe "#run" do
+    it 'should report all the deps in the Gemfile if Gemfile.lock is absent in cyclonedx' do
+      repo = Salus::Repo.new('spec/fixtures/report_ruby_gems/gemfile_only')
+      scanner = Salus::Scanners::ReportRubyGems.new(repository: repo, config: {})
+      scanner.run
+
+      ruby_cyclonedx = Cyclonedx::ReportRubyGems.new(scanner.report)
+      expect(ruby_cyclonedx.build_components_object).to match_array(
+        [
+          {
+            "bom-ref": "pkg:gem/kibana_url",
+            "type": "library",
+            "group": "",
+            "name": "kibana_url",
+            "version": "",
```

Comment:
The Gemfile says `gem 'kibana_url', '~> 1.0'`, so should. you add `~> 1.0` to `version`?

Reply:
Right, ideally the actual version of the Gem should be specified in CycloneDX in semantic versioning format according to the [schema](https://github.com/CycloneDX/specification/blob/master/schema/bom-1.3.schema.json#L291). In the schema it simply requires the version to be a string so we could add a pessimistic version strings as well.